### PR TITLE
Update React docs

### PR DESCRIPTION
**Pull request by MMMalik**
- Renames `v2` document to `Latest` since it covers all major versions starting from v2
- Adds comparison table for integration versions since `v1` up to upcoming `v4` 
- Adds a small section "Data initialization" to cover long overdue https://github.com/ckeditor/ckeditor4-react/issues/264

Closes: https://github.com/ckeditor/ckeditor4-react/issues/264
Touches: https://github.com/ckeditor/ckeditor4-react/issues/284

### CODE REVIEW

**Comment by MMMalik** [comment]
It turns out that it's impossible to build a document **and** remove it from navigation like I intended to do for React integration v1. I've consulted Umberto docs and a few people... Therefore I decided to keep those docs.

**Comment by jacekbogdanski** [comment]
I've changed the upstream branch and rebased branch. This PR should target `master` branch as that version of react integration is not released yet.

**Review by Comandeer — APPROVED** [review]
LGTM! I've just removed mentions of "v2" in the migration guide and added one missing dot.

Other than that, I have one doubt – totally non-blocking for this PR.

File: docs/guide/dev/integration/react/current/README.md [hunk 1]
```diff
@@ -1,23 +1,37 @@
...
 Copyright (c) 2003-2022, CKSource Holding sp. z o.o. All rights reserved.
 For licensing, see LICENSE.md.
 -->
 
-# CKEditor 4 React Integration - v2
+# CKEditor 4 React Integration
 
 <info-box info="">
 	This feature is provided through the <a href="https://www.npmjs.com/package/ckeditor4-react"><code>ckeditor4-react</code> npm package</a>.
+	Document below serves as a reference for public API of `ckeditor4-react` starting from v2.
```
Comment (godai78) [inline docs/guide/dev/integration/react/current/README.md:18]:
```suggestion
	The document below serves as a reference for the public API of `ckeditor4-react` starting from v2.
```

File: docs/guide/dev/integration/react/current/README.md [hunk 2]
```diff
@@ -1,23 +1,37 @@
...
 </info-box>
 
-CKEditor 4 offers a React integration through library described below. By providing a deep integration of CKEditor 4 and React we let you use the native features of the WYSIWYG editor inside your React app. This package is compatible with React version 16.8 or higher.
+CKEditor 4 offers a React integration through library described below. By providing a deep integration of CKEditor 4 and React we let you use the native features of the WYSIWYG editor inside your React app. The latest version of this package is compatible with [React 18](https://reactjs.org/blog/2022/03/29/react-v18.html).
```
Comment (godai78) [inline docs/guide/dev/integration/react/current/README.md:21]:
```suggestion
CKEditor 4 offers a React integration through the library described below. By providing deep integration of CKEditor 4 and React we let you use the native features of the WYSIWYG editor inside your React app. The latest version of this package is compatible with [React 18](https://reactjs.org/blog/2022/03/29/react-v18.html).
```

File: docs/guide/dev/integration/react/current/README.md [hunk 3]
```diff
@@ -128,6 +150,34 @@ const component = (
 );
 ```
 
+### Data initialization
+
+Prop `initData` accepts either string with HTML or JSX. Samples below produce equivalent results.
```
Comment (godai78) [inline docs/guide/dev/integration/react/current/README.md:155]:
```suggestion
Prop `initData` accepts either a string with HTML or JSX. The samples below produce equivalent results.
```

File: docs/guide/dev/integration/react/current/README.md [hunk 4]
```diff
@@ -433,21 +483,21 @@ function Editor() {
...
+The previous version of this {@link guide/dev/integration/react/deprecated/README library} provided `CKEditor` component with an interface almost identical to the current one.
 
-Most notable difference is that `data` prop was replaced by `initData` prop. Previously, passing a new value of `data` prop would trigger a call to `editor.setData` under the hood. In v2 data can be set only once, during editor instantiation (hence `initData`). The rationale behind this change is that `data` prop created false illusion of being the only source of truth for editor's data while in fact the editor's state is most of the time controlled by the editor itself.
+Most notable difference is that `data` prop was replaced by `initData` prop. Previously, passing a new value of `data` prop would trigger a call to `editor.setData` under the hood. In newer versions data can be set only once, during editor instantiation (hence `initData`). The rationale behind this change is that `data` prop created false illusion of being the only source of truth for editor's data while in fact the editor's state is most of the time controlled by the editor itself.
```
Comment (godai78) [inline docs/guide/dev/integration/react/current/README.md:490]:
```suggestion
Most notable difference is that `data` prop was replaced by `initData` prop. Previously, passing a new value of `data` prop would trigger a call to `editor.setData` under the hood. In newer versions data can be set only once, during editor instantiation (hence `initData`). The rationale behind this change is that `data` prop created false illusion of being the only source of truth for editor's data while in fact the editor's state is most of the time controlled by the editor itself.
```
```suggestion
The most notable difference is that `data` prop was replaced by `initData` prop. Previously, passing a new value of `data` prop would trigger a call to `editor.setData` under the hood. In newer versions, data can be set only once, during editor instantiation (hence `initData`). The rationale behind this change is that `data` prop created false illusion of being the only source of truth for the editor's data while in fact, the editor's state is most of the time controlled by the editor itself.
```

**Review by godai78 — APPROVED** [review]

**Review by jacekbogdanski — APPROVED** [review]
Good job guys! 👏🏻

**Pull request merged**